[docs] Create Pickers masked field recipe (#13515)
mui/mui-x: docs/data/date-pickers/custom-field/custom-field.md

{{"demo": "PickerWithAutocompleteField.js", "defaultCodeOpen": false}}

> ### Using a masked `TextField`

Reviewer: Best practice is to avoid backticks in headers, even if you would use them in the body text.

```suggestion
### Using a masked Text Field
```

Author: Oops I missed this one

If you want to use a simple mask approach for the field editing instead of the built-in logic,

> you can replace the field with a masked-based `TextField` built using the [rifm](https://www.npmjs.com/package/rifm) package.

Reviewer 2: slight variation in wording, since 'masked-based' did sound a bit off.

```suggestion
you can replace the default field with a `TextField` using a masked input value built with the [rifm](https://www.npmjs.com/package/rifm) package.
```

> If you want to use a simple mask approach for the field editing instead of the built-in logic,
> you can replace the default field with a Text Field using a masked input value built with the [rifm](https://www.npmjs.com/package/rifm) package.

Reviewer: ```suggestion
If you want to use a simple mask approach for field editing instead of the built-in logic, you can replace the default field with a Text Field using a masked input value built with the [rifm](https://www.npmjs.com/package/rifm) package.
```

Reviewer 3: The line break is there just for readability.
It does not impact DOM.
I would not change anything here. 🤔 
One idea is to maybe link to GH instead of npmjs, WDYT?

Author: For readability of the md file (just to be sure it's clear)

Author: > One idea is to maybe link to GH instead of npmjs, WDYT?

Updated

Reviewer 3: 💯

Reviewer: IMO it doesn't feel more readable with the line break at all—it just looks random, like it was added by mistake. If the sentence is difficult to read because of the length, I would rather see it revised to be shorter.
